### DISCUSSION TITLE
Why saleor preferred to not to use redux at all?

### DISCUSSION THREAD

**Discussion by thisisnitinverma** [saleor/saleor#6152] [General]
I am starting a new project from scratch and I decided to use Django with React. But when I say react there are many developers who use redux as an essential library for state management. 

But why saleor team not using redux and chose apollo as an alternative?

**Comment by maarcingebala** [comment]
Dislaimer - these are only my guesses as I don't do much frontend programming these days ;) Apollo Client is designed to work with GraphQL APIs. It handles communication with API, provides components to handle queries and mutations; it also provides a great client caching mechanism. Probably you could use GraphQL with Redux but that would require a bit more hacking and you would lose the benefits that Apollo Client gives.

**Comment by maarcingebala** [comment]
For a more detailed answer, we would need to ask our @mirumee/saleor-frontend team.

**Reply by thisisnitinverma** [reply]
Thank you so much :) I will wait for your response

**Comment by karolkielecki** [comment]
In addition, I'd like to add that we are pleased with the speed and usage of Apollo in Saleor.
